### PR TITLE
Chore/gatsby2 migration

### LINKED ISSUE #139: Migrate to Gatsby 2
Gatsby 2 is currently in beta but I think it's worth to migrate:

- Better webpack performance
- Smaller bundles
- We can use dynamic imports to lazy-load sections, see #69 
- More future proof

### PR DESCRIPTION
**Pull request by PedroMiguelSS**
This PR migrates the project from [gatsbyjs v1](https://v1.gatsbyjs.org/) to [gatsbyjs v2](https://www.gatsbyjs.org/). 

### Important changes:

- Updated `gatsbyjs` version
- Updated `gatsbyjs` related packages
- Updated `gatsbyjs` plugins' peer dependencies:
   - [gatsby-link](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-link)
   - [gatsby-plugin-react-helmet](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-react-helmet)
- layouts (See [ref](https://www.gatsbyjs.org/docs/migrating-from-v1-to-v2/#remove-or-refactor-layout-components))
   - changed `children` `render prop` to a normal `prop`;
   - moved `layouts/index.js` to `src/shared/components/layout`
   - each page should import and be wrapped by the `layout` component
   - updated `layout` component to receive `className` as a `prop`
- Updated `gatsby-node.js` according its [new API](https://www.gatsbyjs.org/docs/node-apis/) and changed webpack config according webpack 4.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,41 +12,64 @@
...
-    "stylelint": "^9.2.0",
+    "serviceworker-webpack-plugin": "^1.0.1",
+    "stylelint": "^9.7.0",
     "stylelint-config-moxy": "^5.0.0",
     "svg-css-modules-loader": "^1.6.1",
-    "svgo": "^1.0.5",
-    "svgo-loader": "^2.1.0"
+    "svgo-loader": "^2.2.0"
   },
-  "browserslist": [
```
Comment (satazor) [inline package.json]: Where is this defined now?

File: src/pages/404/index.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
-    <p>You just hit a route that does not exist... the sadness.</p>
-  </div>
-)
+import Layout from 'shared/components/layout'
+
+const NotFoundPage = ({ pageContext }) => {
+  return (
+    <Layout intlInfo={ pageContext }>
+      <h1>NOT FOUND</h1>
+      <p>You just hit a route that does not exist... the sadness.</p>
```
Comment (satazor) [inline src/pages/404/index.js]: This is not related to this PR, but these messages should be translated.

File: gatsby-config.js [hunk 3]
```diff
@@ -2,37 +2,9 @@ module.exports = {
...
-      resolve: 'gatsby-plugin-manifest',
-      options: {
-        name: 'JS.IPFS',
-        short_name: 'JS.IPFS',
-        theme_color: '#ffffff',
-        background_color: '#ffffff',
-        display: 'standalone',
-        icons: [
-          {
-            src: 'favicon/android-chrome-192x192.png',
```
Comment (vasco-santos) [inline gatsby-config.js]: What about these? Have you tested?

File: src/shared/components/react-live/live-editor.js [hunk 4]
```diff
@@ -1,31 +1,42 @@
 import React, { Component } from 'react'
-import { LiveProvider, Editor } from 'react-live'
+import { withLive, Editor } from 'react-live'
 
-export default class LiveEditor extends Component {
-  static contextTypes = LiveProvider.childContextTypes
+class LiveEditor extends Component {
+  // static contextTypes = LiveProvider.childContextTypes
```
Comment (satazor) [inline src/shared/components/react-live/live-editor.js]: If we don't need this anymore, remove it.

File: src/service-worker/registration.js [hunk 5]
```diff
@@ -1,9 +1,7 @@
-import runtime from 'serviceworker-webpack-plugin/lib/runtime'
```
Comment (vasco-santos) [inline src/service-worker/registration.js]: If we remove this, we do not need to use the plugin. But we will need to change the webpack config to add manually the service worker file. You can check the plugin codebase to understand how they do it

File: package.json (hunk 1 above)
Comment (PedroMiguelSS) [reply package.json]: I did remove it but should have restored it back! Done now.

File: src/pages/404/index.js (hunk 2 above)
Comment (PedroMiguelSS) [reply src/pages/404/index.js]: Will do it soon.

File: gatsby-config.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (PedroMiguelSS) [reply gatsby-config.js]: I've re-installed the `gatsby-plugin-manifest` to generate an `manifest.webmanifest` automatically on each build.

File: src/service-worker/registration.js (hunk 5 above)
Comment (PedroMiguelSS) [reply src/service-worker/registration.js]: I've already restored the plugin.

[the remark this answers is not in the document]
Comment (PedroMiguelSS) [reply src/service-worker/registration.js]: Done

**Comment by PedroMiguelSS** [comment]
Closed in favor of #192.

**Pull request closed without merging**